src/tty_wrap.cc

```diff
@@ -78,6 +79,20 @@ void TTYWrap::GuessHandleType(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(OneByteString(env->isolate(), type));
 }
 
+void TTYWrap::FdProtected(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  int32_t fd = args[0]->Int32Value();
+  CHECK_GE(fd, 0);
+
+  uv_loop_t* loop = env->event_loop();
+  bool ret = false;
+
+  if (static_cast<unsigned>(fd) < loop->nwatchers &&
+      loop->watchers[fd] != NULL)
```

Comment:
`loop->watchers` and `loop->nwatchers` are UNIX-specific.  This won't compile on Windows.